config/base/derive/src/proxy.rs

```diff
@@ -35,86 +37,115 @@ pub fn impl_proxy(ast: StructWithFields) -> TokenStream {
 pub fn impl_combine(ast: &StructWithFields) -> TokenStream {
     let combine_trait = quote! { ::iroha_config_base::proxy::Combine };
     let name = &ast.ident;
-    let fields = utils::extract_field_idents(&ast.fields);
+    let clauses = ast.fields.iter().map(|field| {
+        let field_name = &field.ident;
+        if field.has_inner {
+            let inner_ty = get_inner_type("Option", &field.ty);
+            quote! {
+                if let Some(other_field) = other.#field_name {
+                    if let Some(this_field) = self.#field_name {
+                        self.#field_name = Some(<#inner_ty as #combine_trait>::combine(this_field, other_field))
+                    } else {
+                        self.#field_name = Some(other_field)
+                    }
+                }
```

Comment:
I'd argue that a `match` statement would make it much more readable. Consider
```suggestion
                match (self.#field_name, other.#field_name) {
                      (Some(this), Some(other)) => Some(<#inner_ty as #combine_trait>::combine(this, other)), 
                      (this, None) => this, 
                      (None, other) => other,
                }
```